core/server/src/main/java/alluxio/proxy/FileSystemClientRestServiceHandler.java

```diff
@@ -0,0 +1,474 @@
+        if (!loadMetadataType.isEmpty()) {
+          listStatusOptions.setLoadMetadataType(LoadMetadataType.valueOf(loadMetadataType));
+        }
+        return mFileSystem.listStatus(new AlluxioURI(path), listStatusOptions);
+      }
+    });
+  }
+
+  /**
+   * @summary mount a UFS path
+   * @param path the alluxio mount point
+   * @param ufsPath the UFS path to mount
+   * @param readOnly whether to make the mount option read only
+   * @param shared whether to make the mount option shared with all Alluxio users
+   * @return the response object
+   */
+  @POST
+  @Path(MOUNT)
+  @ReturnType("java.lang.Void")
+  public Response mount(@QueryParam("path") final String path,
+      @QueryParam("ufsPath") final String ufsPath, @QueryParam("readOnly") final Boolean readOnly,
+      @QueryParam("shared") final Boolean shared) {
+    return RestUtils.call(new RestUtils.RestCallable<Void>() {
+      @Override
+      public Void call() throws Exception {
+        Preconditions.checkNotNull(path, "required 'path' parameter is missing");
+        Preconditions.checkNotNull(ufsPath, "required 'ufsPath' parameter is missing");
+        MountOptions options = MountOptions.defaults();
+        if (readOnly != null) {
+          options.setReadOnly(readOnly);
+        }
+        if (shared != null) {
+          options.setShared(shared);
+        }
+        mFileSystem.mount(new AlluxioURI(path), new AlluxioURI(ufsPath), options);
+        return null;
+      }
+    });
+  }
+
+  /**
+   * @summary move a path
+   * @param srcPath the source path
+   * @param dstPath the destination path
+   * @return the response object
+   */
+  @POST
+  @Path(RENAME)
+  @ReturnType("java.lang.Void")
+  public Response rename(@QueryParam("srcPath") final String srcPath,
+      @QueryParam("dstPath") final String dstPath) {
+    return RestUtils.call(new RestUtils.RestCallable<Void>() {
+      @Override
+      public Void call() throws Exception {
+        Preconditions.checkNotNull(srcPath, "required 'srcPath' parameter is missing");
+        Preconditions.checkNotNull(dstPath, "required 'dstPath' parameter is missing");
+        mFileSystem.rename(new AlluxioURI(srcPath), new AlluxioURI(dstPath));
+        return null;
+      }
+    });
+  }
+
+  /**
+   * @summary set an attribute
+   * @param path the file path
+   * @param pinned the pinned flag value to use
+   * @param ttl the time-to-live (in seconds) to use
+   * @param persisted the persisted flag value to use
+   * @param owner the file owner
+   * @param group the file group
+   * @param mode the file permission bits
+   * @param recursive whether the attribute should be set recursively
+   * @param ttlAction action to take after TTL is expired
+   * @return the response object
+   */
+  @POST
+  @Path(SET_ATTRIBUTE)
+  @ReturnType("java.lang.Void")
+  public Response setAttribute(@QueryParam("path") final String path,
+      @QueryParam("pinned") final Boolean pinned, @QueryParam("ttl") final Long ttl,
```

Comment:
done the one-per-line ... as for alphabetization, I would prefer to keep the security options grouped